torchinductor/compile_fx.py

```diff
@@ -106,16 +107,27 @@ def compile_fx(
     gm, wrap = python_key_normalize(
         model, example_inputs, decompositions=decompositions
     )
+
     if config.dce:
         gm.graph.eliminate_dead_code()
     if config.debug:
         gm.graph.print_tabular()
-    if cudagraphs is None:
-        cudagraphs = config.triton.cudagraphs
 
     if os.environ.get("TORCHINDUCTOR_CHECK_OPS") == "1":
         wrap(CheckEachNode(gm).run)(*example_inputs)
 
+    return compile_fx_inner(gm, example_inputs, wrap=wrap, cudagraphs=cudagraphs)
+
+
+def compile_fx_inner(
+    gm: torch.fx.GraphModule,
+    example_inputs: List[torch.Tensor],
+    wrap=identity,
```

Comment:
It is this:
https://github.com/jansel/torchdynamo/blob/df478179e881c6fc17e91a4fe075659a29407777/torchdynamo/optimizations/python_key.py#L126

It adds all the module parameters via `*params_flat` in the call.

Basically this is a lower overhead version of `autograd.Function`.  Rather than creating new `autograd.Function`, it gives you a python closure.